### DISCUSSION TITLE
Add text formatting capabilities to neobasic ( TAB, GETCURSOR )

### DISCUSSION THREAD

**Discussion by erlendnagel** [paulscottrobson/neo6502-firmware#255] [Implemented Ideas]
Neobasic is lacking some formatting commands that exist in some other basics, like PRINT USING, TAB() and SPC(). It is not too hard to create these as a small prcedure though (e.g. https://www.atarimagazines.com/compute/issue49/420_1_Atari_TAB.php ), but what would really make that easier is to have a command that gives the current cursor position (e.g. GETCURSORX() and GETCURSORY() , although the memory position to PEEK for that information would certainly also do, but is sadly not yet documented.

**Comment by paulscottrobson** [comment]
It is an Firmware call, you can't actually PEEK() anything. It's worth considering though.

**Comment by paulscottrobson** [comment]
This is the nearest to peeking. It's a message passing API, so you ask the firmware where the cursor is and wait for the reply. You don't have to wait for most things.
```
'
'	Read cusros position
'
cls:print:print:print "*********";
'
' Wait for any previous messages to be handled
'
while peek($FF00) <> 0:wend
'
' Do command 2,13 (read cursor position). Writing non-zero to $FF00 triggers it.
'
poke $FF01,13
poke $FF00,2
'
' Wait for result
'
while peek($FF00) <> 0:wend
'
' Read from parameters and print
'
x = peek($FF04)
y = peek($FF05)
print:print "At ",x,y
```
